### PR TITLE
add error check of aes_encrypt()

### PR DESCRIPTION
**Pull request by Hxinrong**
### Contribution description

sys/random/fortuna/fortuna.c:add error check of aes_encrypt()
I'm not sure which exact value should be returned when aes_encrypt() is not on success. I just return the return value of it. 

### Issues/PRs references

Addresses parts of #15524

### CODE REVIEW

**Review by benpicco — APPROVED** [review]
this looks good too :)

File: sys/random/fortuna/fortuna.c [hunk 1]
```diff
@@ -67,7 +67,10 @@ static int fortuna_generate_blocks(fortuna_state_t *state, uint8_t *out,
     }
 
     for (size_t i = 0; i < blocks; i++) {
-        aes_encrypt(&cipher, state->gen.counter.bytes, out + (i * 16));
+        res = aes_encrypt(&cipher, state->gen.counter.bytes, out + (i * 16));
+        if (res != 1) {
+            return res;
```
Comment (basilfx) [inline sys/random/fortuna/fortuna.c:72]: I wrote this implementation some time ago. I think the return code should be negative, and I'm not sure about `res` being negative.

**Review by basilfx — CHANGES_REQUESTED** [review]

**Comment by fjmolinas** [comment]
ping @Hxinrong! :)

File: sys/random/fortuna/fortuna.c (hunk 1 above)
Comment (PeterKietzmann) [reply sys/random/fortuna/fortuna.c:72]: Frme [the doc](https://doc.riot-os.org/aes_8h.html#aaca6a1300bb44f1513a28de5de8f9774):

>    1 on success 
>    A negative value if the cipher key cannot be expanded with the AES key schedule 

`fortuna_generate_blocks` is an internal function so it should be fine to return the negative value. *But* maybe an "internal" documentation of that function would be beneficial.

**Comment by PeterKietzmann** [comment]
@basilfx you put a change request on this PR. What did you want changed?

**Comment by basilfx** [comment]
@PeterKietzmann I think that it should not `return res` but `return -3` (or another negative value) if `aes_encrypt != 1`.

The documentation states that `aes_encrypt` either returns 1 or a negative value. Returning `res` could be a problem if `aes_encrypt` would return something else (due to an API change, another implementation, etc).

**Comment by Teufelchen1** [comment]
@basilfx could you give it another look?

**Review by benpicco — APPROVED** [review]
I think this should just blow an `assert()` since nothing about this should depend on the run-time state.

But then again I don't expect anyone to use this over any of the more 'slim' RNGs, so it probably doesn't matter ¯\\_(ツ)_/¯

**Pull request merged**